### PR TITLE
Closes #2839 Remove Age Verify plugin compatibility files

### LINKED ISSUE #2839: Compatibility with Age Verify no longer required and can be removed
We have a custom compatibility with [Age Verify plugin](https://wordpress.org/plugins/age-verify/) here: https://github.com/wp-media/wp-rocket/blob/master/inc/3rd-party/plugins/age-verify.php

During a test, David and I noticed that this is no longer needed because the plugin by itself adds `DONOTCACHEPAGE` until age is verified - https://plugins.trac.wordpress.org/browser/age-verify/trunk/includes/class-age-verify.php#L233

Video - https://youtu.be/PaOm9I3wXKM

Steps:

- Installing the plugin will add `age-verified` as a mandatory cooking in WP Rocket's config file
- Delete the mandatory cookie
- Visit any page and the age verify notice will be displayed. 
- No cache will be generated until the notice is accepted.

### PR DESCRIPTION
**Pull request by jorgemartine00**
## Description

- Adds functions from **age-verify.php** to `inc/deprecated/3.8.php`
- Deletes **age-verify.php**
- Removes `require` for age-verify.php from inc/3rd-party/3rd-party.php

Fixes #2839

## Type of change

- [x] Enhancement (non-breaking change which changes functionality)

## Is the solution different from the one proposed during the grooming?
No

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: inc/deprecated/3.8.php [hunk 1]
```diff
@@ -278,3 +278,59 @@ function rocket_deactivate_specify_image_dimensions_with_layerslider() {
...
+
+/**
+ * Conflict with Age Verify: don't cache pages until the age-verified cookie is set
+ *
+ * @since 2.7
+ */
+if ( class_exists( 'Age_Verify' ) && defined( 'Age_Verify::SLUG' ) ) :
+	add_filter( 'rocket_htaccess_mod_rewrite', '__return_false', 18 );
+	add_filter( 'rocket_cache_mandatory_cookies', 'rocket_add_cache_mandatory_cookie_for_age_verify' );
+endif;
```
Comment (remyperona) [inline inc/deprecated/3.8.php]: You don't need to keep that part, we only want to keep the deprecated functions to make sure there is no errors

File: inc/deprecated/3.8.php [hunk 2]
```diff
@@ -278,3 +278,59 @@ function rocket_deactivate_specify_image_dimensions_with_layerslider() {
...
+
+/**
+ * Add age-verified to the list of mandatory cookies
+ *
+ * @since 2.7
```
Comment (remyperona) [inline inc/deprecated/3.8.php:288]:
```suggestion
 * @since 3.8.6 deprecated
 * @since 2.7
```

File: inc/deprecated/3.8.php [hunk 3]
```diff
@@ -278,3 +278,59 @@ function rocket_deactivate_specify_image_dimensions_with_layerslider() {
...
+ */
+function rocket_add_cache_mandatory_cookie_for_age_verify( $cookies ) {
+	$cookies[] = 'age-verified';
+	return $cookies;
+}
+
+/**
+ * Add age-verified cookie when we activate the plugin
+ *
+ * @since 2.7
```
Comment (remyperona) [inline inc/deprecated/3.8.php:304]:
```suggestion
 * @since 3.8.6 deprecated
 * @since 2.7
```

File: inc/deprecated/3.8.php [hunk 4]
```diff
@@ -278,3 +278,59 @@ function rocket_deactivate_specify_image_dimensions_with_layerslider() {
...
+	add_filter( 'rocket_htaccess_mod_rewrite', '__return_false', 18 );
+	add_filter( 'rocket_cache_mandatory_cookies', 'rocket_add_cache_mandatory_cookie_for_age_verify' );
+
+	// Update the WP Rocket rules on the .htaccess file.
+	flush_rocket_htaccess();
+
+	// Regenerate the config file.
+	rocket_generate_config_file();
+}
+add_action( 'activate_age-verify/age-verify.php', 'rocket_activate_age_verify', 11 );
```
Comment (remyperona) [inline inc/deprecated/3.8.php]:
```suggestion
```

File: inc/deprecated/3.8.php [hunk 5]
```diff
@@ -278,3 +278,59 @@ function rocket_deactivate_specify_image_dimensions_with_layerslider() {
...
+function rocket_deactivate_age_verify() {
+	remove_filter( 'rocket_cache_mandatory_cookies', 'rocket_add_cache_mandatory_cookie_for_age_verify' );
+
+	// Update the WP Rocket rules on the .htaccess file.
+	flush_rocket_htaccess();
+
+	// Regenerate the config file.
+	rocket_generate_config_file();
+}
+add_action( 'deactivate_age-verify/age-verify.php', 'rocket_deactivate_age_verify', 11 );
```
Comment (remyperona) [inline inc/deprecated/3.8.php]:
```suggestion
```

File: inc/deprecated/3.8.php [hunk 6]
```diff
@@ -278,3 +278,59 @@ function rocket_deactivate_specify_image_dimensions_with_layerslider() {
...
+
+	// Regenerate the config file.
+	rocket_generate_config_file();
+}
+add_action( 'activate_age-verify/age-verify.php', 'rocket_activate_age_verify', 11 );
+
+/**
+ * Remove age-verified cookie when we deactivate the plugin
+ *
+ * @since 2.7
```
Comment (Tabrisrp) [inline inc/deprecated/3.8.php:325]:
```suggestion
 * @since 3.8.6 deprecated
 * @since 2.7
```

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Review by piotrbak — APPROVED** [review]
Worked as expected:
[testrail-report-133.pdf](https://github.com/wp-media/wp-rocket/files/6004715/testrail-report-133.pdf)

**Pull request merged**